test/integration_test/basic_test.go

```diff
@@ -307,6 +308,61 @@ func BasicInstall(tc *types.TestCase) func(*testing.T) {
 	}
 }
 
+func TestBasicInstall(t *testing.T) {
+	testCase := &types.TestCase{
+		TestName: "BasicInstall",
+		TestSpec: func(t *testing.T) interface{} {
+			objects, err := ci_utils.ParseSpecs("storagecluster/storagecluster-with-all-components.yaml")
+			require.NoError(t, err)
+			cluster, ok := objects[0].(*corev1.StorageCluster)
+			require.True(t, ok)
+			cluster.Name = "test-basic-stc"
+			cluster.Namespace = ci_utils.PxNamespace
+			return cluster
+		},
+		TestFunc: BasicInstallAetosDemo,
+	}
+	BasicInstallAetosDemo(testCase)(t)
+}
+
+func BasicInstallAetosDemo(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+		dash.TestCaseBegin("BasicInstall", "Deploying and validating StorageCluster", "", nil)
+
+		// Construct StorageCluster
+		aetos.LogInfoD("Constructing StorageCluster %s", cluster.Name)
+		err := ci_utils.ConstructStorageCluster(cluster, ci_utils.PxSpecGenURL, ci_utils.PxSpecImages)
+		dash.VerifySafely(err == nil, true,
+			fmt.Sprintf("Constructed StorageCluster %s? %v", cluster.Name, err))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Deploy PX and validate
+		aetos.LogInfoD("Deploying & Validating StorageCluster %s", cluster.Name)
+		_, err = ci_utils.DeployAndValidateStorageClusterAetosDemo(cluster, ci_utils.PxSpecImages)
+		dash.VerifySafely(err == nil, true,
+			fmt.Sprintf("Deployed and validated StorageCluster %s? %v", cluster.Name, err))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		aetos.LogInfoD("Uninstalling & Validating StorageCluster %s", cluster.Name)
+		err = ci_utils.UninstallAndValidateStorageClusterAetosDemo(cluster)
+		dash.VerifySafely(err == nil, true,
+			fmt.Sprintf("Uninstalled and validate StorageCluster %s? %v", cluster.Name, err))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer dash.TestCaseEnd()
+		defer dash.TestSetEnd()
```

Comment:
How about adding `defer` statements immediately after `TestCaseBegin` and `TestSetBegin`, or next to their respective counterparts? This ensures that if an error triggers `t.Fatal(err)`, causing the test to fail and halt immediately, the `defer` functions will still execute, as they would have been set prior to the point of failure.

Reply:
main_test.go already has TestSetEnd() called. This defer is to defer ending the testSet in case there are other tests in the Focus list. I will remove the defer for TestCaseEnd, good catch!